ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/validator/caching/CachingDamlLedgerStateReaderSpec.scala

```diff
@@ -1,99 +0,0 @@
-// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
-// SPDX-License-Identifier: Apache-2.0
-
-package com.daml.ledger.validator.caching
-
-import com.daml.caching.WeightedCache
-import com.daml.ledger.participant.state.kvutils.DamlKvutils.{DamlStateKey, DamlStateValue}
-import com.daml.ledger.participant.state.kvutils.caching.`Message Weight`
-import com.daml.ledger.validator.ArgumentMatchers.{anyExecutionContext, seqOf}
-import com.daml.ledger.validator.DefaultStateKeySerializationStrategy
-import com.daml.ledger.validator.caching.CachingDamlLedgerStateReaderSpec._
-import com.daml.ledger.validator.reading.DamlLedgerStateReader
-import org.mockito.{ArgumentMatchersSugar, MockitoSugar}
-import org.scalatest.Inside
-import org.scalatest.matchers.should.Matchers
-import org.scalatest.wordspec.AsyncWordSpec
-
-import scala.concurrent.Future
-
-class CachingDamlLedgerStateReaderSpec
-    extends AsyncWordSpec
-    with Matchers
-    with Inside
-    with MockitoSugar
-    with ArgumentMatchersSugar {
-  "read" should {
-    "record read keys" in {
```

Comment:
Do you intend to implement this functionality in the oem-integration-kit only?

Reply:
Yes, as a decorator implementing `StateReader` and tracking read keys.